Source/BoundaryConditions/WarpXFieldBoundaries.cpp

```diff
@@ -76,3 +76,8 @@ void WarpX::ApplyBfieldBoundary (const int lev, PatchType patch_type, DtType a_d
         }
     }
 }
+
+void WarpX::ApplyRhofieldBoundary (const int lev, MultiFab* rho)
+{
+    if (PEC::isAnyBoundaryPEC()) PEC::ApplyPECtoRhofield(rho, lev);
```

Comment:
We should make sure the fine vs coarse patches are properly treated.

Reply:
The level info is currently used only to extract the correct domain box for a given level. Do we need to do more in order to treat refined patches correctly? If that's the case, it might be worth adding a warning or something like that.

Reply:
Thank you for reminding me about this. I think with the latest commit we now correctly handle the coarse versus fine patches.